### ISSUE TITLE
Databricks: ON DELETE NO ACTION ON UPDATE NO ACTION fails to parse

### ISSUE THREAD

**Issue by cchambers-rdi** [sqlfluff/sqlfluff#7949]
### Search before asking

- [x] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

When I attempt to parse the following using the databricks dialect, I get an error about an unparseable section:

``` sql
CREATE TABLE catalog.silver.child (
    parent_key BIGINT NOT NULL,
    CONSTRAINT fk_child_parent FOREIGN KEY (parent_key)
    REFERENCES catalog.silver.parent (parent_key)
    ON DELETE NO ACTION ON UPDATE NO ACTION
) USING DELTA
```
Parsing with the sparksql dialect works fine.

If I reverse the constraint order like this, the databricks dialect will parse it fine:

``` sql
CREATE TABLE catalog.silver.child (
    parent_key BIGINT NOT NULL,
    CONSTRAINT fk_child_parent FOREIGN KEY (parent_key)
    REFERENCES catalog.silver.parent (parent_key)
    ON UPDATE NO ACTION ON DELETE NO ACTION
) USING DELTA
```

### Expected Behaviour

The SQL should parse with the databricks dialect.

### Observed Behaviour

I get a parse error:

```
== [sqlfluff_fk_ddl_mve.sql] FAIL
L:   1 | P:  35 |  PRS | Line 1, Position 35: Found unparsable section: '(\n
                       | parent_key BIGINT NOT NULL,\n    CO...'
```

### How to reproduce

Create a SQL file with the following contents:

``` sql
CREATE TABLE catalog.silver.child (
    parent_key BIGINT NOT NULL,
    CONSTRAINT fk_child_parent FOREIGN KEY (parent_key)
    REFERENCES catalog.silver.parent (parent_key)
    ON DELETE NO ACTION ON UPDATE NO ACTION
) USING DELTA
```
Parse it in the databricks dialect: `sqlfluff lint file.sql --dialect databricks`

### Dialect

Databricks

### Version

4.2.0

### Configuration

[tool.sqlfluff.core]
dialect = "databricks"
templater = "python"
sql_file_exts = ".sql"
output_line_length = 88
max_line_length = 88
max_parse_depth = 500
ignore_paths = ["docs/**"]

[tool.sqlfluff.templater.python.context]
bronze_catalog = "bronze_catalog"
working_catalog = "working_catalog"
catalog = "catalog"
source_catalog = "source_catalog"

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
